### PR TITLE
[FLINK-5344] relax spec for requested ruby version so the docs can be…

### PR DESCRIPTION
**Pull request by alpinegizmo**
With this update, the docs can be built with any version of ruby from 2.0 onward -- I've tested with ruby 2.0 in our dockerized jekyll, and with ruby 2.3 on ubuntu 16.04. Otherwise the run.sh script in docs/docker ran into problems.

The json gem needed to be explicitly pulled in because in early versions of ruby 2.0 it wasn't yet in the ruby stdlib.

### CODE REVIEW

File: docs/build_docs.sh [hunk 1]
```diff
@@ -46,8 +46,8 @@ DOCS_DST=${DOCS_SRC}/content
 JEKYLL_CMD="build"
 
 # if -p flag is provided, serve site on localhost
-# -i is like -p, but incremental (which has some issues, but is very fast)
-while getopts ":p:i" opt; do
+# -i is like -p, but incremental (only rebuilds the modified file)
+while getopts "pi" opt; do
```
Comment (mxm) [inline docs/build_docs.sh]: Does this fix the `-p` argument? In the master, only the `-i` argument is working.

File: docs/Gemfile [hunk 2]
```diff
@@ -17,13 +17,14 @@
 ################################################################################
 
 source 'https://rubygems.org'
-ruby '~> 2.3.0'
+ruby '~> 2'
```
Comment (mxm) [inline docs/Gemfile]: When we upgraded this from `~> 1.9`, the nightly Buildbot stopped working: https://ci.apache.org/builders/flink-docs-master/builds/557/steps/Flink%20docs/logs/stdio

File: docs/build_docs.sh (hunk 1 above)
Comment (alpinegizmo) [inline docs/build_docs.sh]: Yes, this fixes the -p argument.

**Comment by alpinegizmo** [comment]
Max, this latest commit allows one to take advantage of ruby 2.0 if it is available, but is backwards compatible to 1.9. Hopefully this will satisfy everyone. The only issue I can see is that if one runs the build_docs script with ruby 2.x then Gemfile.lock will be updated with newer versions which probably should not be checked in.

**Comment by alpinegizmo** [comment]
I've concluded this was the wrong approach. I'll open another pull request to deal with these issues.

**Pull request closed without merging**